packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx

```diff
@@ -0,0 +1,282 @@
+'use client'
+
+import { FC, useEffect, useMemo, useState } from 'react'
+
+import { useNotification } from '@baseapp-frontend/utils'
+
+import { Divider, Grid, Typography } from '@mui/material'
+import { Box, Theme, useMediaQuery } from '@mui/system'
+import { AddressElement, useElements, useStripe } from '@stripe/react-stripe-js'
+import { useQueryClient } from '@tanstack/react-query'
+
+import useStripeHook from '../../hooks/useStripeHook'
+import ConfirmationSubscriptionModal from '../ConfirmationSubscriptionModal'
+import PaymentDropdown from '../paymentDropDown'
+import { CheckoutProps } from '../types'
+import { ProductContainer, StyledLoadingButton } from './styled'
+
+const Checkout: FC<CheckoutProps> = ({
+  checkoutCustomerId,
+  paymentMethods,
+  product,
+  isLoadingMethods,
+  handleSetupSuccess,
+}) => {
+  const isMobile = useMediaQuery<Theme>((theme) => theme.breakpoints.down('sm'))
+  const [selectedPaymentMethodId, setSelectedPaymentMethodId] = useState<string>('')
+  const [isAddCardModalOpen, setIsAddCardModalOpen] = useState(false)
+  const [isConfirmCardPaymentProcessing, setIsConfirmCardPaymentProcessing] = useState(false)
+  const [orderNumber, setOrderNumber] = useState<string | null>(null)
+  const { sendToast } = useNotification()
+  const [isRetry, setIsRetry] = useState<boolean>(false)
+  const [confirmationModalOpen, setConfirmationModalOpen] = useState<boolean>(false)
+  const [hasMounted, setHasMounted] = useState<boolean>(false)
+
+  const elements = useElements()
+  const stripe = useStripe()
+  const queryClient = useQueryClient()
+
+  const { useCreationSubscription } = useStripeHook()
+  const { mutate: createSubscription, isPending: isCreatingSubscription } =
+    useCreationSubscription()
+
+  const selectedMethod = useMemo(
+    () => paymentMethods.find((pm) => pm.id === selectedPaymentMethodId),
+    [selectedPaymentMethodId, paymentMethods],
+  )
+
+  useEffect(() => {
```

Comment:
do we need this effect? 
Does `hasMounted` mean when we have `paymentMethods` fetched? Is there a point for loading the checkout component without having the payment methods fetched? Maybe we can simplify the logic a bit here

Reply:
The current flow is:
When the component mounts, we initially select the default payment method.
If a new payment method is added afterward, we select that one instead of reverting to the default.
This logic is intended to select the default payment method only once, on the initial mount of the component. 
I can add a comment to explain better, but if you have better suggestion would be great